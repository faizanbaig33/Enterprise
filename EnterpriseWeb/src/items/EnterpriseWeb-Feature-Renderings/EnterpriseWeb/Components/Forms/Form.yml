---
ID: "9092277e-6395-49eb-8170-cdb4613a6185"
Parent: "ac4ace0c-5fef-4491-9cb6-934ba1525734"
Template: "04646a89-996f-4ee7-878a-ffdbf1f0ef0d"
Path: /sitecore/layout/Renderings/Feature/EnterpriseWeb/Components/Forms/Form
SharedFields:
- ID: "037fe404-dd19-4bf7-8e30-4dadf68b27b0"
  Hint: componentName
  Value: Form
- ID: "1a7c85e5-dc0b-490d-9187-bb1dbcb4c72f"
  Hint: Datasource Template
  Value: /sitecore/templates/Feature/EnterpriseWeb/Forms/Form
- ID: "b0b15510-b138-470e-8f33-8da2e228aafe"
  Hint: Rendering Contents Resolver
  Value: "{80CBBB79-4BAF-4E54-A49F-A3AADE5E6034}"
- ID: "b5b27af1-25ef-405c-87ce-369b3a004016"
  Hint: Datasource Location
  Value: "query:@site/data/forms"
Languages:
- Language: en
  Versions:
  - Version: 1
    Fields:
    - ID: "25bed78c-4957-4165-998a-ca1b52f67497"
      Hint: __Created
      Value: 20210711T103042Z
    - ID: "8cdc337e-a112-42fb-bbb4-4143751e123f"
      Hint: __Revision
      Value: "77432fa8-17da-4f73-8599-e8472a1c7d96"
    - ID: "badd9cf9-53e0-4d0c-bcc0-2d784c282f6a"
      Hint: __Updated by
      Value: |
        sitecore\Admin
    - ID: "d9cf14b1-fa16-4ba6-9288-e8a174d4d522"
      Hint: __Updated
      Value: 20230228T193047Z
