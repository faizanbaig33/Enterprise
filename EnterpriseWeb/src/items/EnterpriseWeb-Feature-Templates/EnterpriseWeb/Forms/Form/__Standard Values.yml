---
ID: "c9a1ebdf-c1e8-45fa-9c60-ff021e5c3a65"
Parent: "38f8e205-c367-494c-877e-57344f0f0ffd"
Template: "38f8e205-c367-494c-877e-57344f0f0ffd"
Path: /sitecore/templates/Feature/EnterpriseWeb/Forms/Form/__Standard Values
SharedFields:
- ID: "1172f251-dad4-4efb-a329-0c63500e4f1e"
  Hint: __Masters
  Value: "{C4442680-1100-4626-A469-ACC318E25CBF}"
- ID: "b03569b1-1534-43f2-8c83-bd064b7d782c"
  Hint: __Renderers
  Value: "{9092277E-6395-49EB-8170-CDB4613A6185}"
Languages:
- Language: en
  Versions:
  - Version: 1
    Fields:
    - ID: "25bed78c-4957-4165-998a-ca1b52f67497"
      Hint: __Created
      Value: 20230216T192830Z
    - ID: "52807595-0f8f-4b20-8d2a-cb71d28c6103"
      Hint: __Owner
      Value: |
        sitecore\Admin
    - ID: "5dd74568-4d4b-44c1-b513-0af5f4cda34f"
      Hint: __Created by
      Value: |
        sitecore\Admin
    - ID: "8cdc337e-a112-42fb-bbb4-4143751e123f"
      Hint: __Revision
      Value: "cdab456e-7b30-42ce-9563-ce95b125e0ba"
    - ID: "badd9cf9-53e0-4d0c-bcc0-2d784c282f6a"
      Hint: __Updated by
      Value: |
        sitecore\Admin
    - ID: "d9cf14b1-fa16-4ba6-9288-e8a174d4d522"
      Hint: __Updated
      Value: 20230216T194118Z
