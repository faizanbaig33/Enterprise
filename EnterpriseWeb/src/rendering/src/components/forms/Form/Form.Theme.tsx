// Lib
import { ThemeFile } from 'lib/context/ThemeContext';

export const FormTheme = (): ThemeFile => {
  return {
    aw: {
      classes: {
        form: 'block max-w-lg p-6',
        steps: '',
      },
    },
    rba: {
      classes: {
        form: 'block max-w-lg p-6',
        steps: '',
      },
    },
  };
};
