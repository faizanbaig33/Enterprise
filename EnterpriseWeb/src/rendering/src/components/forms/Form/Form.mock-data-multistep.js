const defaultData = {
    rendering: {
      dataSource: "sampledatasource",
      componentName: "Form",
    },
    uid: "24d4ed52-6782-403a-aa81-56d62c5fbcba",
    componentName: "Form",
    dataSource: "{C4F28335-FB0D-4DCD-98B5-42AC46F4A2B5}",
    fields:{
      children: [
        {
          id: "8dc3ff7f-f96b-4597-805a-a0bc3b4be44a",
          displayName: "Form Page",
          name: "Form Page",
          templateId: "c4442680-1100-4626-a469-acc318e25cbf",
          templateName: "Page",
          url: "/Data/Forms/Test-Form/Form-Page",
          fields: {
            includeInSteps: {
              value: true
            },
            label: {
              value: "Step 1"
            },
            children: [
              {
                id: "acbf0a81-7c9e-430c-b484-ada66a2c1c96",
                displayName: "Headline",
                name: "Headline",
                templateId: "78e78e19-8b37-4637-8c04-9de221572d0d",
                templateName: "Headline",
                url: "/Data/Forms/Test-Form/Form-Page/Headline",
                fields: {
                  headlineLevel: {
                    id: "0f556f3a-bbad-4aa5-952d-b79003b39cd6",
                    url: "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2",
                    name: "Heading 2",
                    displayName: "Heading 2",
                    fields: {
                      Value: {
                        value: "h2"
                      }
                    },
                    templateId: "d2923fee-da4e-49be-830c-e27764dfa269",
                    templateName: "Enum"
                  },
                  superscriptCTA: {
                    value: {
                      href: ""
                    }
                  },
                  headlineText: {
                    value: "This is the Test Form"
                  },
                  children: []
                }
              },
              {
                id: "be4e5ca5-5297-4932-9fd9-265d024e1c1a",
                displayName: "Form Intro",
                name: "Form Intro",
                templateId: "1e258863-055b-4c0d-b775-59541b0a90fd",
                templateName: "Paragraph",
                url: "/Data/Forms/Test-Form/Form-Page/Form-Intro",
                fields: {
                  text: {
                    value: "Here is some rich text form intro copy"
                  },
                  children: []
                }
              },
              {
                id: "7896f125-b041-4edc-b237-04dcd6d49eca",
                displayName: "First Name",
                name: "First Name",
                templateId: "4fd88841-d725-4815-8b03-6ea12c956c18",
                templateName: "Short Text",
                url: "/Data/Forms/Test-Form/Form-Page/First-Name",
                fields: {
                  maxLength: {
                    value: 255
                  },
                  minLength: {
                    value: 0
                  },
                  placeholderText: {
                    value: "Enter your first name"
                  },
                  defaultValue: {
                    value: ""
                  },
                  label: {
                    value: "First Name"
                  },
                  required: {
                    value: true
                  },
                  validations: [],
                  fieldMapping: {
                    id: "e17a5ecb-bbe1-4d26-9a55-625d19b375ee",
                    url: "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Forms/Field-Mappings/Enabled-Plus/First-Name",
                    name: "First Name",
                    displayName: "First Name",
                    fields: {
                      Value: {
                        value: "FirstName"
                      }
                    },
                    templateId: "d2923fee-da4e-49be-830c-e27764dfa269",
                    templateName: "Enum"
                  },
                  children: []
                }
              },
              {
                id: "efd220ce-75db-41aa-9714-1cbebef4054f",
                displayName: "Windows Problems",
                name: "Windows Problems",
                templateId: "62d234ca-1c65-4699-8bfe-339c28d92761",
                templateName: "Long Text",
                url: "/Data/Forms/Test-Form/Form-Page/Windows-Problems",
                fields: {
                  defaultValue: {
                    value: "This is a default value"
                  },
                  maxLength: {
                    value: null
                  },
                  minLength: {
                    value: 0
                  },
                  placeholderText: {
                    value: ""
                  },
                  rows: {
                    value: 5
                  },
                  label: {
                    value: "Long Text"
                  },
                  required: {
                    value: false
                  },
                  validations: [],
                  fieldMapping: {
                    id: "2954da41-3a5c-4658-8edb-a597fa5f9a6d",
                    url: "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Forms/Field-Mappings/Enabled-Plus/Windows-Problems",
                    name: "Windows Problems",
                    displayName: "Windows Problems",
                    fields: {
                      Value: {
                        value: "WindowsProblems"
                      }
                    },
                    templateId: "d2923fee-da4e-49be-830c-e27764dfa269",
                    templateName: "Enum"
                  },
                  children: []
                }
              },
              {
                id: "3866d094-39b9-4eb7-937b-7eaf67192c8c",
                displayName: "Go Forward",
                name: "Go Forward",
                templateId: "6dad441f-58d9-40a0-a119-c8d990b0484d",
                templateName: "Button",
                url: "/Data/Forms/Test-Form/Form-Page/Go-Forward",
                fields: {
                  navigationStep: {
                    id: "289fe73e-e947-4b59-a53a-6bfbf997dd45",
                    url: "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Forms/Navigation-Steps/Next",
                    name: "Next",
                    displayName: "Next",
                    fields: {
                      Value: {
                        value: "1"
                      }
                    },
                    templateId: "d2923fee-da4e-49be-830c-e27764dfa269",
                    templateName: "Enum"
                  },
                  label: {
                    value: "Go Forward"
                  },
                  children: []
                }
              }
            ]
          }
        },
        {
          id: "b99d23e5-cbbe-4e4f-9b97-bc61e2e47ca4",
          displayName: "Form Page 2",
          name: "Form Page 2",
          templateId: "c4442680-1100-4626-a469-acc318e25cbf",
          templateName: "Page",
          url: "/Data/Forms/Test-Form/Form-Page-2",
          fields: {
            includeInSteps: {
              value: true
            },
            label: {
              value: "Step 2"
            },
            children: [
              {
                id: "d2cc647b-a912-467d-9ace-aa01e3cffa2d",
                displayName: "Headline",
                name: "Headline",
                templateId: "78e78e19-8b37-4637-8c04-9de221572d0d",
                templateName: "Headline",
                url: "/Data/Forms/Test-Form/Form-Page-2/Headline",
                fields: {
                  headlineLevel: {
                    id: "0f556f3a-bbad-4aa5-952d-b79003b39cd6",
                    url: "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2",
                    name: "Heading 2",
                    displayName: "Heading 2",
                    fields: {
                      Value: {
                        value: "h2"
                      }
                    },
                    templateId: "d2923fee-da4e-49be-830c-e27764dfa269",
                    templateName: "Enum"
                  },
                  superscriptCTA: {
                    value: {
                      href: ""
                    }
                  },
                  headlineText: {
                    value: "This is the Test Form"
                  },
                  children: []
                }
              },
              {
                id: "0094025b-dccc-4edc-9a33-2c7505d3fb4e",
                displayName: "Form Intro",
                name: "Form Intro",
                templateId: "1e258863-055b-4c0d-b775-59541b0a90fd",
                templateName: "Paragraph",
                url: "/Data/Forms/Test-Form/Form-Page-2/Form-Intro",
                fields: {
                  text: {
                    value: "Here is some rich text form intro copy"
                  },
                  children: []
                }
              },
              {
                id: "c2d73a68-28e0-4f56-be03-0606228b7dea",
                displayName: "Short Text",
                name: "Short Text",
                templateId: "4fd88841-d725-4815-8b03-6ea12c956c18",
                templateName: "Short Text",
                url: "/Data/Forms/Test-Form/Form-Page-2/Short-Text",
                fields: {
                  maxLength: {
                    value: 255
                  },
                  minLength: {
                    value: 0
                  },
                  placeholderText: {
                    value: "Enter your first name"
                  },
                  defaultValue: {
                    value: ""
                  },
                  label: {
                    value: "First Name"
                  },
                  required: {
                    value: true
                  },
                  validations: [],
                  fieldMapping: null,
                  children: []
                }
              },
              {
                id: "fc70d11c-9a41-42f7-b95b-514276972ebb",
                displayName: "Long Text",
                name: "Long Text",
                templateId: "62d234ca-1c65-4699-8bfe-339c28d92761",
                templateName: "Long Text",
                url: "/Data/Forms/Test-Form/Form-Page-2/Long-Text",
                fields: {
                  defaultValue: {
                    value: ""
                  },
                  maxLength: {
                    value: null
                  },
                  minLength: {
                    value: 0
                  },
                  placeholderText: {
                    value: ""
                  },
                  rows: {
                    value: 5
                  },
                  label: {
                    value: "Long Text"
                  },
                  required: {
                    value: false
                  },
                  validations: [],
                  fieldMapping: null,
                  children: []
                }
              },
              {
                id: "432798cc-208b-42bf-91bc-814ef4ea1937",
                displayName: "Go Back",
                name: "Go Back",
                templateId: "6dad441f-58d9-40a0-a119-c8d990b0484d",
                templateName: "Button",
                url: "/Data/Forms/Test-Form/Form-Page-2/Go-Back",
                fields: {
                  navigationStep: {
                    id: "2847b5eb-514d-4b93-8c7a-4d94d592a9e1",
                    url: "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Forms/Navigation-Steps/Previous",
                    name: "Previous",
                    displayName: "Previous",
                    fields: {
                      Value: {
                        value: "-1"
                      }
                    },
                    templateId: "d2923fee-da4e-49be-830c-e27764dfa269",
                    templateName: "Enum"
                  },
                  label: {
                    value: "Go Back"
                  },
                  children: []
                }
              },
              {
                id: "28d29f88-6350-4f3f-9f9a-de47cb77c5b5",
                displayName: "Go Forward",
                name: "Go Forward",
                templateId: "6dad441f-58d9-40a0-a119-c8d990b0484d",
                templateName: "Button",
                url: "/Data/Forms/Test-Form/Form-Page-2/Go-Forward",
                fields: {
                  navigationStep: {
                    id: "289fe73e-e947-4b59-a53a-6bfbf997dd45",
                    url: "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Forms/Navigation-Steps/Next",
                    name: "Next",
                    displayName: "Next",
                    fields: {
                      Value: {
                        value: "1"
                      }
                    },
                    templateId: "d2923fee-da4e-49be-830c-e27764dfa269",
                    templateName: "Enum"
                  },
                  label: {
                    value: "Submit"
                  },
                  children: [
                    {
                      id: "4b9a2c25-324c-48ad-9463-ae2b62db72e7",
                      displayName: "Stand In Submit Action",
                      name: "Stand In Submit Action",
                      templateId: "a87a00b1-e6db-45ab-8b54-636fec3b5523",
                      templateName: "Folder",
                      url: "/Data/Forms/Test-Form/Form-Page-2/Go-Forward/Stand-In-Submit-Action",
                      fields: {}
                    }
                  ]
                }
              }
            ]
          }
        },
        {
          id: "df13811d-79bf-4c9d-a840-d9e802f14840",
          displayName: "Thank You Page",
          name: "Thank You Page",
          templateId: "c4442680-1100-4626-a469-acc318e25cbf",
          templateName: "Page",
          url: "/Data/Forms/Test-Form/Thank-You-Page",
          fields: {
            includeInSteps: {
              value: false
            },
            label: {
              value: "Thank You Page"
            },
            children: [
              {
                id: "45d23a12-fb58-4b36-882e-95c8c5a1acfa",
                displayName: "Thank You Messaging",
                name: "Thank You Messaging",
                templateId: "1e258863-055b-4c0d-b775-59541b0a90fd",
                templateName: "Paragraph",
                url: "/Data/Forms/Test-Form/Thank-You-Page/Thank-You-Messaging",
                fields: {
                  text: {
                    value: "Thank you for submitting!"
                  },
                  children: []
                }
              },
              {
                id: "09a523ec-f156-4431-82e2-05aa3dbd6fbb",
                displayName: "Go Back",
                name: "Go Back",
                templateId: "6dad441f-58d9-40a0-a119-c8d990b0484d",
                templateName: "Button",
                url: "/Data/Forms/Test-Form/Thank-You-Page/Go-Back",
                fields: {
                  navigationStep: {
                    id: "2847b5eb-514d-4b93-8c7a-4d94d592a9e1",
                    url: "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Forms/Navigation-Steps/Previous",
                    name: "Previous",
                    displayName: "Previous",
                    fields: {
                      Value: {
                        value: "-1"
                      }
                    },
                    templateId: "d2923fee-da4e-49be-830c-e27764dfa269",
                    templateName: "Enum"
                  },
                  label: {
                    value: "Go Back"
                  },
                  children: []
                }
              }
            ]
          }
        }
      ]
    }
  };
export default defaultData;
