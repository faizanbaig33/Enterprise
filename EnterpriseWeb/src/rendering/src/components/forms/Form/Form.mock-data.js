const defaultData = {
    rendering: {
      dataSource: "sampledatasource",
      componentName: "Form",
    },
    uid: "24d4ed52-6782-403a-aa81-56d62c5fbcba",
    componentName: "Form",
    dataSource: "{C4F28335-FB0D-4DCD-98B5-42AC46F4A2B5}",
    fields:{
      children: [
        {
          id: "eb07fcde-fd5a-4e51-b3bc-9426a1ce5ed6",
          displayName: "Page 1",
          name: "Page 1",
          templateId: "c4442680-1100-4626-a469-acc318e25cbf",
          templateName: "Page",
          url: "/Data/Forms/Single-Form/Page-1",
          fields: {
            includeInSteps: {
              value: false
            },
            label: {
              value: "Page 1"
            },
            children: [
              {
                id: "8ce3ee66-51d3-4125-b673-64ec81351a28",
                displayName: "Headline",
                name: "Headline",
                templateId: "78e78e19-8b37-4637-8c04-9de221572d0d",
                templateName: "Headline",
                url: "/Data/Forms/Single-Form/Page-1/Headline",
                fields: {
                  headlineLevel: {
                    id: "0f556f3a-bbad-4aa5-952d-b79003b39cd6",
                    url: "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2",
                    name: "Heading 2",
                    displayName: "Heading 2",
                    fields: {
                      Value: {
                        value: "h2"
                      }
                    },
                    templateId: "d2923fee-da4e-49be-830c-e27764dfa269",
                    templateName: "Enum"
                  },
                  superscriptCTA: {
                    value: {
                      href: ""
                    }
                  },
                  headlineText: {
                    value: "Headline"
                  }
                }
              },
              {
                id: "755fd39f-c5dc-44e2-92dd-f1c57261463d",
                displayName: "Paragraph",
                name: "Paragraph",
                templateId: "1e258863-055b-4c0d-b775-59541b0a90fd",
                templateName: "Paragraph",
                url: "/Data/Forms/Single-Form/Page-1/Paragraph",
                fields: {
                  text: {
                    value: "Lorem ipsum stuff here"
                  }
                }
              },
              {
                id: "1bdb2818-0846-4998-8a68-a9d0401a9be2",
                displayName: "Short Text",
                name: "Short Text",
                templateId: "4fd88841-d725-4815-8b03-6ea12c956c18",
                templateName: "Short Text",
                url: "/Data/Forms/Single-Form/Page-1/Short-Text",
                fields: {
                  maxLength: {
                    value: 5
                  },
                  minLength: {
                    value: 2
                  },
                  placeholderText: {
                    value: ""
                  },
                  defaultValue: {
                    value: "Default value"
                  },
                  label: {
                    value: "Short Text"
                  },
                  required: {
                    value: true
                  },
                  validations: [],
                  fieldMapping: null
                }
              },
              {
                id: "2bdb2818-0846-4998-8a68-a9d0401a9be2",
                displayName: "Email address",
                name: "Email address",
                templateId: "A1D62E74-51C3-4BDD-8547-1ECE87422EC9",
                templateName: "Email",
                url: "/Data/Forms/Single-Form/Page-1/Email-address",
                fields: {
                  maxLength: {
                    value: null
                  },
                  minLength: {
                    value: null
                  },
                  placeholderText: {
                    value: ""
                  },
                  label: {
                    value: "Email"
                  },
                  required: {
                    value: true
                  },
                  validations: [],
                  fieldMapping: null
                }
              },
              {
                id: "181f6d65-9671-4fd8-8484-d34a1bcdaacc",
                displayName: "Long Text",
                name: "Long Text",
                templateId: "62d234ca-1c65-4699-8bfe-339c28d92761",
                templateName: "Long Text",
                url: "/Data/Forms/Single-Form/Page-1/Long-Text",
                fields: {
                  defaultValue: {
                    value: "Other default value"
                  },
                  maxLength: {
                    value: null
                  },
                  minLength: {
                    value: 0
                  },
                  placeholderText: {
                    value: "Placeholder text"
                  },
                  rows: {
                    value: 5
                  },
                  label: {
                    value: "Long Text"
                  },
                  required: {
                    value: true
                  },
                  validations: [],
                  fieldMapping: null
                }
              },
              {
                id: "d96e2463-c348-4876-92eb-aef28923b606",
                displayName: "Hidden",
                name: "Hidden",
                templateId: "b01e7cae-d802-4ae5-9742-bdcda546554b",
                templateName: "Hidden",
                url: "/Data/Forms/Single-Form/Page-1/Hidden",
                fields: {
                  defaultValue: {
                    value: "2"
                  },
                  fieldMapping: {
                    id: "2f5aa35e-eee8-4cfc-a267-47791a562cfb",
                    url: "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Forms/Field-Mappings/Enabled-Plus/Form-Type",
                    name: "Form Type",
                    displayName: "Form Type",
                    fields: {
                      Value: {
                        value: "FormType"
                      }
                    },
                    templateId: "d2923fee-da4e-49be-830c-e27764dfa269",
                    templateName: "Enum"
                  }
                }
              },
              {
                id: "a2c73b02-265e-4b21-932d-c465eb479578",
                displayName: "Button",
                name: "Button",
                templateId: "6dad441f-58d9-40a0-a119-c8d990b0484d",
                templateName: "Button",
                url: "/Data/Forms/Single-Form/Page-1/Button",
                fields: {
                  navigationStep: {
                    id: "b6185a15-c782-4ddf-8ad4-21620a8fe9d4",
                    url: "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Forms/Navigation-Steps/Submit",
                    name: "Submit",
                    displayName: "Submit",
                    fields: {
                      Value: {
                        value: "0"
                      }
                    },
                    templateId: "d2923fee-da4e-49be-830c-e27764dfa269",
                    templateName: "Enum"
                  },
                  label: {
                    value: "Button"
                  }
                }
              }
            ]
          }
        }
      ]
    }
  };
export default defaultData;
