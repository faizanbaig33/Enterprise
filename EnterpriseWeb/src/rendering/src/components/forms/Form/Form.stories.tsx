// Global
import { Story, Meta } from '@storybook/react';
// Local
import Form, { FormProps } from './Form';
import defaultData from './Form.mock-data';
import multiStepData from './Form.mock-data-multistep';

export default {
  title: 'Forms/Form',
  component: Form,
} as Meta;
const Template: Story<FormProps> = (props) => <Form {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;

export const multistep = Template.bind({});
multistep.args = multiStepData;
