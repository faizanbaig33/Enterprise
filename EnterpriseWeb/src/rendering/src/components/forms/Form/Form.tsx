import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
import classNames from 'classnames';
import { ComponentProps } from 'lib/component-props';
import { FormFieldProps } from 'lib/forms/FormFieldProps';
import { FormEvent, useState } from 'react';
import Page from 'src/helpers/Forms/Structure/Page';
import Steps from 'src/helpers/Forms/Structure/Steps';
import { FieldState, FormsContext } from 'lib/forms/FormContext';
import { FormTheme } from './Form.Theme';
import { useTheme } from 'lib/context/ThemeContext';
import { translateFieldMappings, isFormValid } from 'lib/forms/FormActions';

export type FormProps = ComponentProps & {
  fields: {
    children: Array<FormFieldProps>;
  };
  classes?: string;
};

const Form = (props: FormProps): JSX.Element => {
  const [pageIndex, setPageIndex] = useState(0);
  const { themeData } = useTheme(FormTheme());

  const [fieldStates, setFieldStates] = useState({} as Record<string, FieldState>);

  if (!props.fields) {
    return <></>;
  }

  const handleSubmit = (e: FormEvent) => {
    // Prevent the browser from reloading the page
    e.preventDefault();
    if (isFormValid(fieldStates)) {
      const formModel = translateFieldMappings(props, fieldStates);
      console.log(formModel);
      console.log('submitted');
    }
  };

  const steps = props.fields?.children.filter((x) => x.fields?.includeInSteps?.value);

  return (
    <FormsContext.Provider
      value={{ pageIndex, setPageIndex, fieldStates, setFieldStates, formProps: props }}
    >
      <div className={classNames(themeData.classes.form, props.classes)}>
        <form onSubmit={handleSubmit} noValidate>
          {steps.length > 1 && <Steps steps={steps} />}
          <div className="pages">
            {props.fields?.children.map((child: FormFieldProps, index: number) => {
              return <Page {...child} key={index} classes={index === pageIndex ? '' : 'hidden'} />;
            })}
          </div>
        </form>
      </div>
    </FormsContext.Provider>
  );
};

export default withDatasourceCheck()<FormProps>(Form);
