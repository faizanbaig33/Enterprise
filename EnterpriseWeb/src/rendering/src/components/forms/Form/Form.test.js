// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import Form from './Form';
import defaultData from './Form.mock-data';

it('renders correctly', () => {
  const component = snapshot(Form, { componentProps: defaultData });
});
