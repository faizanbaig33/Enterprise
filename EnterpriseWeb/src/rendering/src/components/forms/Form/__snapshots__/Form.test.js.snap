// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="block max-w-lg p-6"
>
  <form
    novalidate=""
  >
    <div
      class="pages"
    >
      <div
        class=""
      >
        <div
          class="relative mb-s"
        >
          <div
            class="items-top font-sans  text-theme-text text-sm-m md:text-m font-heavy"
          >
            <h2>
              Headline
            </h2>
            
          </div>
        </div>
        <div
          class="relative mb-s"
        >
          <div
            class="body-copy"
          >
            <div>
              Lorem ipsum stuff here
            </div>
          </div>
        </div>
        <div
          class="relative mb-s"
          data-te-input-wrapper-init="true"
        >
          <label
            class="text-body font-regular flex mb-xxs"
            for="1bdb2818-0846-4998-8a68-a9d0401a9be2"
          >
            Short Text
             
            *
          </label>
          <input
            class="flex h-12 w-full rounded-none border border-gray outline-none focus:border-black py-3 pr-3 pl-2"
            id="1bdb2818-0846-4998-8a68-a9d0401a9be2"
            maxlength="5"
            minlength="2"
            name="Short Text"
            placeholder=""
            required=""
            type="text"
            value="Default value"
          />
        </div>
        <div
          class="relative mb-s"
          data-te-input-wrapper-init="true"
        >
          <label
            class="text-body font-regular flex mb-xxs"
            for="2bdb2818-0846-4998-8a68-a9d0401a9be2"
          >
            Email
             
            *
          </label>
          <input
            class="flex h-12 w-full rounded-none border border-gray outline-none focus:border-black py-3 pr-3 pl-2"
            id="2bdb2818-0846-4998-8a68-a9d0401a9be2"
            name="Email address"
            placeholder=""
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="relative mb-s"
          data-te-input-wrapper-init="true"
        >
          <label
            class="text-body font-regular flex mb-xxs"
            for="181f6d65-9671-4fd8-8484-d34a1bcdaacc"
          >
            Long Text
             
            *
          </label>
          <textarea
            class="peer block min-h-[auto] w-full rounded-none border border-gray outline-none focus:border-black py-3 pr-3 pl-2"
            id="181f6d65-9671-4fd8-8484-d34a1bcdaacc"
            minlength="0"
            name="Long Text"
            placeholder="Placeholder text"
            required=""
            rows="5"
          >
            Other default value
          </textarea>
        </div>
        <div
          class="relative mb-s"
          data-te-input-wrapper-init="true"
        >
          <input
            id="d96e2463-c348-4876-92eb-aef28923b606"
            name="Hidden"
            readonly=""
            type="hidden"
            value="2"
          />
        </div>
        <div
          class="relative mb-s"
        >
          <button
            name="Button"
            type="submit"
          >
            Button
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;
